### PR TITLE
[GH-5951] Update Authentication > OpenID

### PR DESCRIPTION
**Pull request by matthew-src**
#### Summary

This PR updates the OpenID section of the Authentication configuration settings.

#### Ticket Link

This PR partly addresses https://github.com/mattermost/docs/issues/5951.

### CODE REVIEW

**Comment by cwarnermm** [comment]
Hi @matthew-src! Please see [my comments](https://github.com/mattermost/docs/pull/6131#issuecomment-1379368898) for details on how to address the merge conflicts in this PR. And please don't hesitate to reach out if I can be of assistance. Happy to help with this!

**Comment by matthew-src** [comment]
I think that the conflicts are resolved, and that these metadata descriptions should be reviewed:

- Office 365 Open ID Directory tenant ID
- Office 365 Open ID Client ID (references Google account; this error is in the current documentation)
- Office 365 Open ID Client secret (also references Google account)

**Comment by matthew-src** [comment]
Thank you, @cwarnermm. I updated those settings such that they would conform to the description in the table, as I did in PR https://github.com/mattermost/docs/pull/6131.

**Comment by cwarnermm** [comment]
You're doing a great job adding contextual links to the configuration setting documentation, and those additions are very helpful to system admins looking for more information about a given config.

Reviewing the preview, I want to call out a few minor things about the readability of the tables:

image
- The content in brackets is currently forcing the width of the column to expand, and the second column to become less readable. 
- While the table columns auto-adjust, you can force the table width by adding a hard return earlier in the sentence. You can even break up sentences mid-link as long as the text wrapped to the next line left-aligns correctly. In this case it's bullets, so it'll be a hanging indent to preserve.
- The word "settings" is currently a link, but "settings" applies to both contexts in a given line. I recommend updating the link syntax to remove "settings" from the link itself so that it simply displays as text. This change will also help links that contain brackets.

image
- There's considerable whitespace in the left column which is forcing the right column to be less readable. This is likely due to extra spaces inserted into the left column. You can force that column's width as noted above.
- Really like the additional links to help guide the user's experience. What do you think of moving all "see also" links to a subsequent row. See ``configure/environment-configuration-settings.html#managed-resource-paths`` as an example. This would introduce more consistency across the config settings pages and draw more attention to the default value details.

**Review by cwarnermm — COMMENTED** [review]
Thanks, @matthew-src! I've provided feedback inline.

**Comment by matthew-src** [comment]
Hi @cwarnermm, I have done some work based on your feedback. I realized that the additional whitespace was caused in part by the long link addresses in the tables, although the addresses themselves are invisible when the HTML is rendered. My solution is to move the links to another row, and to shorten both the visible link text and the address. I implemented this change for the GitLab section in the latest commit. Please let me know what you think.

File: source/configure/authentication-configuration-settings.rst [hunk 1]
```diff
@@ -2176,21 +2176,39 @@ Select OpenID Connect service provider
 
 *Available in legacy Enterprise Edition E20*
 
-Choose whether OpenID Connect can be used for account creation and login. Options include:
++-------------------------------------------------------------------------------+---------------------------------------------------------------+
+| Use this setting to enable OpenID Connect, with these options:                | - System Config path: **Authentication > OpenID Connect**     | 
+|                                                                               | - ``config.json`` setting: N/A                                |
+| - **Do not allow login via an OpenID provider**                               | - Environment variable: N/A                                   |
+| - **GitLab** `(settings) <#gitlab-openid-settings>`__                         |                                                               |
```
Comment (cwarnermm) [inline source/configure/authentication-configuration-settings.rst:2183]: I think we need a verb in front of "settings" as a clear call to action, as "see settings".

**Comment by cwarnermm** [comment]
Thanks, @matthew-src! I've provided feedback inline to update the link text.

**Comment by matthew-src** [comment]
Hi @cwarnermm, I would resolve this conflict by deleting the section where the links were changed, as this will be replaced entirely by the new table. Is that acceptable? Also, should the links in the table be full URLs instead of anchors?

**Comment by cwarnermm** [comment]
Hi @matthew-src. I agree with your approach to resolve the merge conflict on this PR. And all internal links between pages of the product docs site should be relative and not absolute because relative links are more robust/less prone to break when content is reorganized.

You'll see absolute URLs creep in because Sphinx supports a number of relative link types, each with their own required syntax, that contributors find overly complex and cumbersome to support.

File: source/configure/authentication-configuration-settings.rst [hunk 2]
```diff
@@ -2379,43 +2438,67 @@ This value is the ID of the application's AAD directory.
...
+OpenID Connect (other) settings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. include:: ../_static/badges/ent-pro-cloud-selfhosted.rst
+  :start-after: :nosearch:
+
+Enable OpenID Connect authentication with other service providers
+'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+.. include:: ../_static/badges/ent-pro-cloud-selfhosted.rst
```
Comment (cwarnermm) [inline source/configure/authentication-configuration-settings.rst:2500]: this duplicate line should be removed

File: source/configure/authentication-configuration-settings.rst [hunk 3]
```diff
@@ -2379,43 +2438,67 @@ This value is the ID of the application's AAD directory.
...
+  :start-after: :nosearch:
```
Comment (cwarnermm) [inline source/configure/authentication-configuration-settings.rst:2501]: this duplicate line should be removed

**Review by cwarnermm — COMMENTED** [review]
Thanks, @matthew-src. I've flagged 2 lines that should be removed as duplicates.

**Comment by matthew-src** [comment]
@cwarnermm Got it!

**Comment by cwarnermm** [comment]
My mistake! The content I thought was missing isn't missing at all; I was looking at section headers. Approving.

**Review by cwarnermm — APPROVED** [review]
Thanks, @matthew-src!

**Pull request merged**